lambda/matching.ml

```diff
@@ -1044,29 +1044,34 @@ type ('args, 'row) pattern_matching = {
   default : Default_environment.t
 }
 
-type args = (lambda * let_kind) list
+type 'a arg = {
+  arg : 'a;
+  str : let_kind;
```

Comment:
Minor nitpick: `str` is quite a vague field name. It looks like a shortcut for `structure`, but it might not be so.

Reply:
`str` is the variable name that was previously used in the code in various places, so reusing it makes the diff substantially less noisy thanks to punning

Reply:
(but I can also rename the field and then rebase my later PRs with the new field name if that is your preference)

Reply:
I think that it is mostly a waste of time, but the reviewer is king. I took 21 minutes to rename `str` into `binding_kind`.